solutions/devsprint-luiz-zytkowski-4/src/test/io/devpass/creditcard/domain/CreditCardOperationServiceTest.kt

```diff
@@ -0,0 +1,137 @@
+        val creditCardDAO = mockk<ICreditCardDAO> {
+            every { getById(any()) } returns creditCardReference
+        }
+        val creditCardOperationDAO = mockk<ICreditCardOperationDAO> {
+            every {
+                listByPeriod(
+                    creditCardId = "",
+                    month = 1,
+                    year = 2000
+                )
+            } returns creditCardOperationReference
+        }
+        val creditCardOperationService =
+            CreditCardOperationService(creditCardDAO, creditCardInvoiceDAO, creditCardOperationDAO)
+        val result = creditCardOperationService.listByPeriod("", month = 1, year = 2000)
+        assertEquals(creditCardOperationReference, result)
+    }
+
+    @Test
+    fun `Should return a BusinessRuleException when year is invalid`() {
+        val creditCardReference = getRandomCreditCard()
+        val creditCardInvoiceDAO = mockk<ICreditCardInvoiceDAO>()
+        val creditCardDAO = mockk<ICreditCardDAO> {
+            every { getById(any()) } returns creditCardReference
+        }
+        val creditCardOperationDAO = mockk<ICreditCardOperationDAO> {
+            every {
+                listByPeriod(
+                    creditCardId = "",
+                    month = 12,
+                    year = 0
+                )
+            } throws BusinessRuleException("You must provide a valid year to list operations")
+        }
+        val creditCardOperationService =
+            CreditCardOperationService(creditCardDAO, creditCardInvoiceDAO, creditCardOperationDAO)
+        assertThrows<BusinessRuleException> {
+            creditCardOperationService.listByPeriod("", 12, 0)
+        }
+    }
+
+    @Test
+    fun `Should return a BusinessRuleException when month is invalid`() {
+        val creditCardReference = getRandomCreditCard()
+        val creditCardInvoiceDAO = mockk<ICreditCardInvoiceDAO>()
+        val creditCardDAO = mockk<ICreditCardDAO> {
+            every { getById(any()) } returns creditCardReference
+        }
+        val creditCardOperationDAO = mockk<ICreditCardOperationDAO> {
+            every {
+                listByPeriod(
+                    creditCardId = "",
+                    month = 13,
+                    year = 2000
+                )
+            } throws BusinessRuleException("You must provide a valid month to list operations")
+        }
+        val creditCardOperationService =
+            CreditCardOperationService(creditCardDAO, creditCardInvoiceDAO, creditCardOperationDAO)
+        assertThrows<BusinessRuleException> {
+            creditCardOperationService.listByPeriod("", 13, 2000)
+        }
+    }
+
+    @Test
+    fun `Should return a EntityNotFoundException when credit card is null`() {
+        val creditCardInvoiceDAO = mockk<ICreditCardInvoiceDAO>()
+        val creditCardDAO = mockk<ICreditCardDAO> {
+            every { getById(any()) }  throws EntityNotFoundException("Credit card not found with ID")
```

Comment:
`returns null` em vez de `throws EntityNotFoundException("Credit card not found with ID")`